libparsec/crates/client/tests/unit/client/recovery.rs

```diff
@@ -0,0 +1,119 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use crate::{create_device_from_recovery, DeviceInfo};
+
+use libparsec_client_connection::{AuthenticatedCmds, ProxyConfig};
+use libparsec_platform_device_loader::{get_default_key_file, load_device};
+use libparsec_tests_fixtures::prelude::*;
+use libparsec_types::prelude::*;
+
+use super::utils::client_factory;
+
+fn assert_device(info: &DeviceInfo, expected_device: impl TryInto<DeviceID>, env: &TestbedEnv) {
+    let (certif, _) = env.get_device_certificate(expected_device);
+
+    let DeviceInfo {
+        id,
+        device_label,
+        created_on,
+        created_by,
+    } = info;
+    p_assert_eq!(id, &certif.device_id);
+    let expected_device_label = match &certif.device_label {
+        MaybeRedacted::Real(device_label) => device_label,
+        MaybeRedacted::Redacted(_) => unreachable!(),
+    };
+    p_assert_eq!(device_label, expected_device_label);
+    p_assert_eq!(created_on, &certif.timestamp);
+    let expected_created_by = match &certif.author {
+        CertificateSignerOwned::User(author) => Some(author),
+        CertificateSignerOwned::Root => None,
+    };
+    p_assert_eq!(created_by.as_ref(), expected_created_by);
+}
+
+#[parsec_test(testbed = "coolorg", with_server)]
+async fn ok(env: &TestbedEnv) {
+    // Part 1: create recovery device
+    let alice = env.local_device("alice@dev1");
+    let client = client_factory(&env.discriminant_dir, alice).await;
+
+    let config = client.config.clone();
+    let config_dir = config.config_dir.clone();
+    let recovery_device_label = DeviceLabel::try_from("recovery").unwrap();
+    let (passphrase, data) = client
+        .export_recovery_device(recovery_device_label.clone())
+        .await
+        .unwrap();
+    client.refresh_workspaces_list().await.unwrap();
+
+    // but why do you test workspace count ?
```

Comment:
Shouldn't we compare `alice` with `recovery_device` and `new_device` on the fields that should have been preserved instead ? (again with an explicit destructuring to be explicit about the fields that are not compared in order to be resilient against newly added fields)

Reply:
I wasn't able to compare recovery device. The associated local device is not exposed anymore.

But I compare the new device below ^^